stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -345,6 +346,7 @@ public static void prepareOutboundResponse(Context context, HttpCarbonMessage in
 
         HttpUtil.addHTTPSessionAndCorsHeaders(context, inboundRequestMsg, outboundResponseMsg);
         HttpUtil.enrichOutboundMessage(outboundResponseMsg, outboundResponseStruct);
+        HttpUtil.setHeaderToResponseMsg(inboundRequestMsg, outboundResponseMsg);
```

Comment:
We don't have to set the server header to the response message as a header unless it is is different from the one set to the listenerConfiguration. So we can remove this